src/python/pants/option/options.py

```diff
@@ -86,11 +86,9 @@ def complete_scopes(cls, scope_infos):
     # components) we can replace this line with `for si in scope_infos:`, because it will
```

Comment:
Is this TODO still valid?

Reply:
While I can see methods such as `Optionable.validate_scope_name_component()`, it's not clear to me how to enforce that any optionable must have a valid `options_scope` at all times -- see [the bottom of `optionable.py`](https://github.com/pantsbuild/pants/blob/83104a088155cb5c17a9840905abb35c7f2ab75d/src/python/pants/option/optionable.py#L72). Enforcing a valid scope name component seems doable (which would then allow removal of this TODO), but I haven't delved into how that would be done.